### ISSUE TITLE
Handle missing elevation data

### ISSUE THREAD

**Issue by nrenner** [GIScience/Leaflet.Heightgraph#108]
Heightgraph currently doesn't handle cases where part or all of the data points have coordinates, but no elevation/altitude value.

This results in incomplete graphs and NaN errors/warnings, see comparison below and the Plunker console:

current | expected 
-- | --
[Plunker](http://plnkr.co/edit/BWSX4Y9XjCdyLCiy?preview) | [Plunker](http://plnkr.co/edit/0jLCagjFvf2o0HNZd09M?p=preview), see also [Elevation#84](https://github.com/MrMufflon/Leaflet.Elevation/pull/84)
![image](https://user-images.githubusercontent.com/1092030/99651738-4a2e4600-2a57-11eb-8540-99b028a332d4.png) | ![image](https://user-images.githubusercontent.com/1092030/73591679-d89a0100-44f1-11ea-92ab-03f2ab4417d2.png)

Diagrams shown are variants of the same elevation profile with missing values at different positions:
1. original
2. middle (left of second segment)
3. left
4. right
5. all missing

Reasons for missing elevation data e.g. are using CGIAR CSI SRTM that has no data beyond 60 degrees north and south and outside shorelines, see abrensch/brouter#137, nrenner/brouter-web#203.

The suggested solution would be to show missing elevation values as gaps by using a d3 [defined](https://github.com/d3/d3-shape/blob/master/README.md#area_defined) accessor function, e.g. [here](https://github.com/GIScience/Leaflet.Heightgraph/blob/9e45353ae01e4389124701bdacb62ee255d3fec1/src/L.Control.Heightgraph.js#L776-L780):
```
d3Area().defined(d => d.altitude !== undefined && d.altitude !== null)
```

I guess I could provide a PR if no one else feels like it.

**Comment by boldtrn** [comment]
Have you considered filling emtpy slots on your side? That way you could interpolate the elevation or fill it with 0 etc., depending on your needs.

I am not sure if this plugin should fix "incorrect"/"missing" data?

**Comment by nrenner** [comment]
A corresponding issue is [Elevation #63](https://github.com/MrMufflon/Leaflet.Elevation/issues/63) with a few upvotes and another PR, so we're not the only ones.

While interpolation or assuming 0 might work for some cases, it will be misleading or just wrong for others. I personally would expect missing elevation data to be reflected in the graph as such.

**Comment by boldtrn** [comment]
Setting it to 0 would result in the same rendering result as you shown above wouldn't it?

The difference would then be that when hovering over the entry that instead of showing 0 there would be something like "not defined"? That said, feel free to propose a PR and check back with @TheGreatRefrigerator if this features is desired for this repo.

**Comment by nrenner** [comment]
> Setting it to 0 would result in the same rendering result as you shown above wouldn't it?

Not quite. The connection would be different and rest would only look the same because this artificial example has a min height of 0 and I disabled extending the elevations bounds for better comparison.

With a more realistic example the differences are:
- the graph outline is flattened when the elevation range is extended to 0
- `defined` cuts the graph off with a vertical line, while with 0 it connects normally with a diagonal line to the next data point
- due to elevation bounds extension by 10%, the graph visibly continues at height 0, making the filling look like an actual height of 0

0 filling:

![image](https://user-images.githubusercontent.com/1092030/102380451-37605000-3fc8-11eb-974f-44c825aa304d.png)

`defined`:

![image](https://user-images.githubusercontent.com/1092030/102380617-68d91b80-3fc8-11eb-8903-16adfd59de6d.png)

**Comment by ghost** [comment]
Could https://github.com/nrenner/brouter-web/issues/854 be related with this?

At the moment, both brouter fdroid and brouter-web show no elevation data in part of west Norway (at least)
